src/renderers/noop/ReactNoop.js

```diff
@@ -20,33 +20,97 @@
 'use strict';
 
 import type { Fiber } from 'ReactFiber';
+import type { HostChildren } from 'ReactFiberReconciler';
 
 var ReactFiberReconciler = require('ReactFiberReconciler');
 
 var scheduledHighPriCallback = null;
 var scheduledLowPriCallback = null;
 
+const TERMINAL_TAG = 99;
+
+type Container = { rootID: number, children: Array<Instance> };
+type Props = { };
+type Instance = { tag: 99, type: string, id: number, children: Array<Instance> };
+
+var instanceCounter = 0;
+
+function recursivelyAppendChildren(flatArray : Array<Instance>, child : HostChildren<Instance>) {
+  if (!child) {
+    return;
+  }
+  if (child.tag === TERMINAL_TAG) {
```

Comment:
I'm not really happy with this model of determining terminalness by testing the host environment's type. I'd like to structure the output so that I always know if the next value is terminal or not.

One thing I could do is skip the propagation of single values up the tree and just output the child for those cases. More to traverse but you have to do it at some point I guess. Then I'd just check if the tag is a host node and if so, then I know the output is terminal.

@spicyj Any ideas?